primefaces/src/main/resources/META-INF/resources/primefaces/forms/forms.selectbooleancheckbox.js

```diff
@@ -110,22 +124,26 @@ PrimeFaces.widget.SelectBooleanCheckbox = PrimeFaces.widget.BaseWidget.extend({
      * @param {boolean} [silent] `true` to suppress triggering event listeners, or `false` otherwise.
      */
     uncheck: function(silent) {
-        if(this.isChecked()) {
-            this.input.prop('checked', false);
-            if(!silent) {
+        if (this.isChecked()) {
+            this.input.prop('checked', false).attr('aria-checked', false);
+            this.box.removeClass('ui-state-active')
+                .children('.ui-chkbox-icon')
+                .removeClass('ui-icon-check')
+                .addClass('ui-icon-blank');
+
+            if (!silent) {
                 this.input.trigger('change');
             }
-            this.input.attr('aria-checked', false);
-            this.box.removeClass('ui-state-active').children('.ui-chkbox-icon').addClass('ui-icon-blank').removeClass('ui-icon-check');
         }
     },
 
     /**
      * Enables this input so that the user can enter a value.
      */
     enable: function() {
-        PrimeFaces.utils.enableInputWidget(this.jq, this.input);
         this.disabled = false;
+        PrimeFaces.utils.enableInputWidget(this.jq, this.input);
+        this.bindEvents();
```

Comment:
I guess this is the crucial part of the fix, right?

Reply:
Yes when you re enable all of the events need to be attached and the ui-state-disabled needs to be removed.